doc/user-guide/data-structures.rst

```diff
@@ -495,6 +497,314 @@ dimension and non-dimension variables:
+inherited dimensions, but DataTree's inheritance is slightly stricter yet
+easier to reason about.
+
+The constraint that this puts on a DataTree is that dimensions and indices that
+are inherited must be aligned with any direct decendent node's existing
+dimension or index.  This allows decendents to use dimensions defined in
+ancestor nodes, without duplicating that information. But as a consequence, if
+a dimension dimension-name is defined in on a node and that same dimension-name
+exists in one of its ancestors, they must align (have the same index and
+size).
+
+Some examples:
+
+.. ipython:: python
+
+    # Set up coordinates
+    time = xr.DataArray(data=["2022-01", "2023-01"], dims="time")
+    stations = xr.DataArray(data=list("abcdef"), dims="station")
+    lon = [-100, -80, -60]
+    lat = [10, 20, 30]
+
+    # Set up fake data
+    wind_speed = xr.DataArray(np.ones((2, 6)) * 2, dims=("time", "station"))
+    pressure = xr.DataArray(np.ones((2, 6)) * 3, dims=("time", "station"))
+    air_temperature = xr.DataArray(np.ones((2, 6)) * 4, dims=("time", "station"))
+    dewpoint = xr.DataArray(np.ones((2, 6)) * 5, dims=("time", "station"))
+    infrared = xr.DataArray(np.ones((2, 3, 3)) * 6, dims=("time", "lon", "lat"))
+    true_color = xr.DataArray(np.ones((2, 3, 3)) * 7, dims=("time", "lon", "lat"))
+
+    dt2 = xr.DataTree.from_dict(
+        {
+            "/": xr.Dataset(
+                coords={"time": time},
+            ),
+            "/weather": xr.Dataset(
+                coords={"station": stations},
+                data_vars={
+                    "wind_speed": wind_speed,
+                    "pressure": pressure,
+                },
+            ),
+            "/weather/temperature": xr.Dataset(
+                data_vars={
+                    "air_temperature": air_temperature,
+                    "dewpoint": dewpoint,
+                },
+            ),
+            "/satellite": xr.Dataset(
+                coords={"lat": lat, "lon": lon},
+                data_vars={
+                    "infrared": infrared,
+                    "true_color": true_color,
+                },
+            ),
+        },
+    )
+    dt2
+
+
+Here there are four different coordinate variables, which apply to variables in the DataTree in different ways:
+
+``time`` is a shared coordinate used by both ``weather`` and ``satellite`` variables
+``station`` is used only for ``weather`` variables
+``lat`` and ``lon`` are only use for ``satellite`` images
+
+Coordinate variables are inherited to descendent nodes, which means that
+variables at different levels of a hierarchical DataTree are always
+aligned. Placing the ``time`` variable at the root node automatically indicates
+that it applies to all descendent nodes. Similarly, ``station`` is in the base
+``weather`` node, because it applies to all weather variables, both directly
+in ``weather`` and in the ``temperature`` sub-tree.
```

Comment:
This could be a good place to show and explain the repr for `dt2['/weather']` or `dt2['/satellite']`, which should show "Inherited coordinates" (after updates from my repr PR merged a few days ago).

Reply:
Could we maybe make an example dataset modelled after the `weather` and `satellite` structure, as a solution to #9437? 

@eni-awowale maybe you know of some data sources that might be useful for that? (If so let's continue that discussion on #9437)

Reply:
Yes, this was stolen directly from the original issue https://github.com/pydata/xarray/issues/9077. so not from real data